### PR TITLE
Custom/cat additional fasta (delocalise rnaseq module)

### PR DESCRIPTION
**Pull request by pinin4fjords**
This is a de-localised version of the module from rna-seq: https://github.com/nf-core/rnaseq/blob/master/modules/local/cat_additional_fasta/main.nf, updated for module standards.

I'm currently looking to reuse various components from rnaseq, so need to de-localise to avoid lots of redundant code.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/custom/catadditionalfasta/environment.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/environment-schema.json
+name: "custom_catadditionalfasta"
+channels:
+  - conda-forge
+  - bioconda
+  - defaults
+dependencies:
+  - "YOUR-TOOL-HERE"
```
Comment (SPPearce) [inline modules/nf-core/custom/catadditionalfasta/environment.yml:9]: What tool does it need?

File: modules/nf-core/custom/catadditionalfasta/main.nf [hunk 2]
```diff
@@ -0,0 +1,24 @@
+process CUSTOM_CATADDITIONALFASTA {
+    tag "$add_fasta"
```
Comment (SPPearce) [inline modules/nf-core/custom/catadditionalfasta/main.nf:2]: This would normally be meta.id

**Review by SPPearce — COMMENTED** [review]
Bit confused as to what this is actually doing.
It takes a fasta and gtf file that already exist, as well as an extra fasta file.
It then makes a gtf file from the new fasta, and then `cat`s it onto the previous fasta/gtf.
Is there not a tool to perform the gtf -> fasta conversion? Could that not be used instead of the python script?

**Comment by pinin4fjords** [comment]
> Is there not a tool to perform the gtf -> fasta conversion? Could that not be used instead of the python script?

Not that I'm immediately aware of. I'm just trying to avoid copying this local thing between related workflows.

**Review by maxulysse — APPROVED** [review]

**Pull request merged**